lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/WorkbasketServiceImpl.java

```diff
@@ -173,38 +173,30 @@ public Workbasket createWorkbasket(Workbasket newWorkbasket)
 
   @Override
   public Workbasket updateWorkbasket(Workbasket workbasketToUpdate)
-      throws NotAuthorizedException, WorkbasketNotFoundException, ConcurrencyException {
+      throws NotAuthorizedException, WorkbasketNotFoundException, ConcurrencyException,
+          InvalidWorkbasketException {
 
     LOGGER.debug("entry to updateWorkbasket(Workbasket = {})", workbasketToUpdate);
 
     taskanaEngine.getEngine().checkRoleMembership(TaskanaRole.BUSINESS_ADMIN, TaskanaRole.ADMIN);
-
-    WorkbasketImpl workbasketImplToUpdate = null;
+    WorkbasketImpl workbasketImplToUpdate = (WorkbasketImpl) workbasketToUpdate;
+    validateNameAndType(workbasketToUpdate);
```

Comment:
I think we could have just re-used the validateWorkbasket method here

Reply:
Possible, I thought it would only be necessary to check those two. Problem could be that according to line 193-197 in method update(...) it seems to be possible that you update a workbasket having Id null or empty (it won't update the id) which thus should not throw an exception if wanted by intention.